drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -298,46 +306,108 @@ impl Transceivers {
+        self.masked_port_op(
+            WriteOp::BitClear,
+            mask.into(),
+            Addr::QSFP_CTRL_EN_H,
+        )
     }
 
     /// Set reset bits per the specified `mask`
-    pub fn set_reset(&self, mask: u32) -> Result<(), FpgaError> {
-        self.masked_port_op(WriteOp::BitSet, mask, Addr::QSFP_CTRL_RESET_H)
+    pub fn set_reset<M: Into<FpgaPortMasks>>(
+        &self,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.masked_port_op(
+            WriteOp::BitSet,
+            mask.into(),
+            Addr::QSFP_CTRL_RESET_H,
+        )
     }
 
     /// Clear reset bits per the specified `mask`
-    pub fn clear_reset(&self, mask: u32) -> Result<(), FpgaError> {
-        self.masked_port_op(WriteOp::BitClear, mask, Addr::QSFP_CTRL_RESET_H)
+    pub fn clear_reset<M: Into<FpgaPortMasks>>(
+        &self,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.masked_port_op(
+            WriteOp::BitClear,
+            mask.into(),
+            Addr::QSFP_CTRL_RESET_H,
+        )
     }
 
     /// Set lpmode bits per the specified `mask`
-    pub fn set_lpmode(&self, mask: u32) -> Result<(), FpgaError> {
-        self.masked_port_op(WriteOp::BitSet, mask, Addr::QSFP_CTRL_LPMODE_H)
+    pub fn set_lpmode<M: Into<FpgaPortMasks>>(
+        &self,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.masked_port_op(
+            WriteOp::BitSet,
+            mask.into(),
+            Addr::QSFP_CTRL_LPMODE_H,
+        )
     }
 
     /// Clear reset bits per the specified `mask`
-    pub fn clear_lpmode(&self, mask: u32) -> Result<(), FpgaError> {
-        self.masked_port_op(WriteOp::BitClear, mask, Addr::QSFP_CTRL_LPMODE_H)
+    pub fn clear_lpmode<M: Into<FpgaPortMasks>>(
+        &self,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.masked_port_op(
+            WriteOp::BitClear,
+            mask.into(),
+            Addr::QSFP_CTRL_LPMODE_H,
+        )
+    }
+
+    /// Initiate an I2C random read on all ports per the specified `mask`.
+    ///
+    /// The maximum value of `num_bytes` is 128.
+    pub fn setup_i2c_read<M: Into<FpgaPortMasks>>(
+        &self,
+        reg: u8,
+        num_bytes: u8,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.setup_i2c_op(true, reg, num_bytes, mask)
+    }
+
+    /// Initiate an I2C write on all ports per the specified `mask`.
+    ///
+    /// The maximum value of `num_bytes` is 128.
+    pub fn setup_i2c_write<M: Into<FpgaPortMasks>>(
+        &self,
+        reg: u8,
+        num_bytes: u8,
+        mask: M,
+    ) -> Result<(), FpgaError> {
+        self.setup_i2c_op(false, reg, num_bytes, mask)
     }
 
     /// Initiate an I2C operation on all ports per the specified `mask`. When
     /// `is_read` is true, the operation will be a random-read, not a pure I2C
     /// read. The maximum value of `num_bytes` is 128.
-    pub fn setup_i2c_op(
+    pub fn setup_i2c_op<M: Into<FpgaPortMasks>>(
```

Comment:
Does this still need to be `pub` given the addition of `setup_i2c_{read,write}`?

Reply:
It did because it was exposed by the Idol API, but I'm going to change that as well.